src/utils.js

```diff
@@ -35,17 +38,43 @@ export function resolvePathFromService(services, name, parentName) {
 
 	let parentAddress = getServiceResolution(services, parentName);
 
-	let urlParts = (/^(([^:\/?#]+):)?(\/\/([^\/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?/g).exec(parentAddress);
+	let urlParts = getRegex().exec(parentAddress);
 
 	return urlParts[1] + urlParts[3] + join(
 		urlParts[5], '../', name
 	);
 }
 
+/**
+ * Given a resolved service url, generate a corresponding absolute path from the original service name.
+ * This is needed to resolve relative paths within the service name.
+ *
+ * @param {String} service The full service name requested to sofe, may include a relative path
+ * @param {String} url The resolved url of the service
+ *
+ * @return {String} A new url with. Possibly the same as the input url.
+ */
+export function getUrlFromService(service, url) {
+	let parts = getRegex().exec(service)[5].split('/');
+	parts = !parts[0] ? parts.slice(1) : parts;
+
+	const path = parts.slice(1).join('/');
```

Comment:
Will there *always* be at least two elements in the array? Even if on line 59 we already sliced out the first element?